src/test/java/io/cdap/plugin/google/drive/etl/ETLTestGoogleSheets.java

```diff
@@ -0,0 +1,276 @@
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class ETLTestGoogleSheets extends HydratorTestBase {
+
+  @ClassRule
+  public static final TestConfiguration CONFIG = new TestConfiguration("explore.enabled", true);
+  protected static Sheets service;
+  protected static Drive drive;
+  protected static String directoryIdentifier;
+  protected static final String GCP_SERVICE_ACCOUNT_PATH = "google.application.credentials.path";
+  protected static final String GCP_SERVICE_ACCOUNT_BASE64_ENCODED = "google.application.credentials.base64.encoded";
+  protected static final String APPLICATION_NAME = "Google Sheets Test";
+  protected static final Schema INPUT_SCHEMA = Schema.recordOf(
+    "input-record",
+    Schema.Field.of("body", Schema.of(Schema.Type.BYTES)),
+    Schema.Field.of("offset", Schema.of(Schema.Type.LONG)));
+
+  protected static final ArtifactSummary APP_ARTIFACT = new ArtifactSummary("data-pipeline", "3.2.0");
+  protected static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+
+
+  @BeforeClass
+  public static void setupClient() throws Exception {
+
+    // base64-encode the credentials, to avoid a commandline-parsing error, since the credentials have dashes in them
+    String property = System.getProperty(GCP_SERVICE_ACCOUNT_BASE64_ENCODED);
+    String serviceAccountCredentials;
+    if (property != null) {
+      serviceAccountCredentials = Bytes.toString(Base64.getDecoder().decode(property));
+    } else {
+      property = Preconditions.checkNotNull(System.getProperty(GCP_SERVICE_ACCOUNT_PATH),
+                                            "The credentials file provided is null. " +
+                                              "Please make sure the path is correct and the file exists.");
+
+      serviceAccountCredentials = new String(Files.readAllBytes(Paths.get(property)), StandardCharsets.UTF_8);
+    }
+
+    final NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+
+    GoogleCredential googleCredential = null;
+    if (serviceAccountCredentials != null) {
+      try (InputStream inputStream = new ByteArrayInputStream(
+        serviceAccountCredentials.getBytes(StandardCharsets.UTF_8))) {
+        googleCredential = GoogleCredential.fromStream(inputStream).createScoped(
+          Collections.singletonList(DriveScopes.DRIVE));
+      }
+    }
+
+    service = new Sheets.Builder(httpTransport, JSON_FACTORY, googleCredential)
+      .setApplicationName(APPLICATION_NAME)
+      .build();
+
+    drive = new Drive.Builder(httpTransport, JSON_FACTORY, googleCredential)
+      .setApplicationName(APPLICATION_NAME)
+      .build();
+
+    //Create the directory
+    File directoryMetadata = new File();
+    directoryMetadata.setName("TestDirectory");
+    directoryMetadata.setMimeType("application/vnd.google-apps.folder");
+    File directory = drive.files().create(directoryMetadata)
+      .setFields("id")
+      .execute();
```

Comment:
@flakrimjusufi it looks like this code is similar on both classes - maybe using a common method would be a cleaner approach.

Reply:
PR updated.